extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -147,12 +141,56 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 				});
 			});
 		} catch (e) {
-			vscode.window.showErrorMessage(localize('openUntitledNotebookError', "Open file {0} as untitled failed: {1}",
+			vscode.window.showErrorMessage(localize('openUntitledNotebookError', "Open untitled notebook {0} as untitled failed: {1}",
 				resource,
 				e instanceof Error ? e.message : e));
 		}
 	}
 
+	async saveJupyterBooks(): Promise<void> {
+		if (this.currentBook.bookPath) {
+			const allFilesFilter = localize('allFiles', "All Files");
+			let filter: any = {};
+			filter[allFilesFilter] = '*';
+			let uris = await vscode.window.showOpenDialog({
+				filters: filter,
+				canSelectFiles: false,
+				canSelectMany: false,
+				canSelectFolders: true,
+				openLabel: localize('labelPickFolder', "Pick Folder")
+			});
+			if (uris && uris.length > 0) {
+				let pickedFolder = uris[0];
+				let destinationUri: vscode.Uri = vscode.Uri.file(path.join(pickedFolder.fsPath, path.basename(this.currentBook.bookPath)));
+				if (destinationUri) {
+					if (await fs.pathExists(destinationUri.fsPath)) {
+						let doReplace = await this.confirmReplace();
+						if (!doReplace) {
+							return undefined;
+						}
+						else {
+							//remove folder if exists
+							await fs.remove(destinationUri.fsPath);
+						}
+					}
+					//make directory for each contribution book.
+					await fs.mkdir(destinationUri.fsPath);
+					await fs.copy(this.currentBook.bookPath, destinationUri.fsPath);
+
+					//remove book from the books
+					let untitledBookIndex: number = this.books.indexOf(this.currentBook);
```

Comment:
This comment could be clearer...why are you doing it? I'd also say more that you're removing it and then readding it so it appears in the correct tree (untitled -> saved)

Reply:
yep, that what am precisely trying to do here.